types/mydecimal_test.go

```diff
@@ -145,6 +145,41 @@ func (s *testMyDecimalSuite) TestToFloat(c *C) {
 	}
 }
 
+func (s *testMyDecimalSuite) TestToHashKey(c *C) {
+	type tcase struct {
+		numbers []string
+	}
+	var dotest = func(c *C, tests []tcase) {
```

Comment:
I think this function is not needed, it looks ugly and reduces some code readability. And we normally won't call this function multiply times. Can we remove this function?